pkg/prometheus/prometheus.go

```diff
@@ -37,37 +28,102 @@ var ResponseSizeBuckets = append(append(append(append(
 	prometheus.LinearBuckets(1000000, 1000000, 5)...),
 )
 
-// WithTelemetry instruments the HTTP server with prometheus
-func WithTelemetry(handler http.Handler) http.HandlerFunc {
-	counter := prometheus.NewCounterVec(
+var (
+	// server metrics
+	counter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Name: "http_requests_total",
 			Help: "A counter for requests to the wrapped handler.",
 		},
-		[]string{"code"},
+		[]string{"code", "method"},
 	)
 
-	duration := prometheus.NewHistogramVec(
+	duration = prometheus.NewHistogramVec(
```

Comment:
TIOLO: This file switches back and forth between "duration" and "latency". Maybe this is better for a follow-up PR, but we refer to "request duration" as "latency" in all the rest of our metrics and documentation. Perhaps consider changing the variable naming in here as well?